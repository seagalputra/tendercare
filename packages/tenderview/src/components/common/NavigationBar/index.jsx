import React from 'react'
import { useHistory } from 'react-router-dom'
import { makeStyles } from '@material-ui/core/styles'
import {
  AppBar,
  Toolbar,
  Typography,
  Button,
  IconButton
} from '@material-ui/core'
import MenuIcon from '@material-ui/icons/Menu'

import { isLoggedIn, clearToken } from 'utils/sessions'

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1
  },
  navigation: {
    boxShadow: 'none'
  },
  menuButton: {
    marginRight: theme.spacing(2)
  },
  title: {
    flexGrow: 1
  }
}))

const NavigationBar = () => {
  const classes = useStyles()
  const history = useHistory()

  const handleLogout = () => {
    clearToken()
    history.push('/')
  }

  return (
    <header className={classes.root}>
      <AppBar
        className={classes.navigation}
        position="static"
        color="transparent"
      >
        <Toolbar>
          <IconButton
            edge="start"
            className={classes.menuButton}
            color="inherit"
            aria-label="menu"
          >
            <MenuIcon />
          </IconButton>
          <Typography variant="h6" className={classes.title} color="inherit">
            Tendercare
          </Typography>
          {isLoggedIn() ? (
            <Button color="inherit" onClick={handleLogout}>
              Keluar
            </Button>
          ) : null}
        </Toolbar>
      </AppBar>
    </header>
  )
}

export default NavigationBar
