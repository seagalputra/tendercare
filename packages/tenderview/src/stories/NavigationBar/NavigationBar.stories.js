import React from 'react'
import NavigationBar from 'components/common/NavigationBar'

export default {
  title: 'Navigation Bar',
  component: NavigationBar,
  excudeStories: /.*Data$/
}

export const Default = () => <NavigationBar />
