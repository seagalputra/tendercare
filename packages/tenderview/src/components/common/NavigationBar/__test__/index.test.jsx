import React from 'react'
import { shallow } from 'enzyme'
import NavigationBar from '..'

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: jest.fn()
  })
}))

describe('NavigationBar', () => {
  it('renders without crashing', () => {
    const component = shallow(<NavigationBar />)
    expect(component).toMatchSnapshot()
  })
})
