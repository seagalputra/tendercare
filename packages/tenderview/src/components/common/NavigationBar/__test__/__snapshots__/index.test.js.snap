// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar renders without crashing 1`] = `
<nav
  className="flex items-center justify-between flex-wrap bg-transparent p-6 border-b"
>
  <div
    className="flex items-center flex-shrink-0 text-gray-800 mr-6"
  >
    <svg
      className="fill-current h-8 w-8 mr-2"
      height="54"
      viewBox="0 0 54 54"
      width="54"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13.5 22.1c1.8-7.2 6.3-10.8 13.5-10.8 10.8 0 12.15 8.1 17.55 9.45 3.6.9 6.75-.45 9.45-4.05-1.8 7.2-6.3 10.8-13.5 10.8-10.8 0-12.15-8.1-17.55-9.45-3.6-.9-6.75.45-9.45 4.05zM0 38.3c1.8-7.2 6.3-10.8 13.5-10.8 10.8 0 12.15 8.1 17.55 9.45 3.6.9 6.75-.45 9.45-4.05-1.8 7.2-6.3 10.8-13.5 10.8-10.8 0-12.15-8.1-17.55-9.45-3.6-.9-6.75.45-9.45 4.05z"
      />
    </svg>
    <span
      className="font-semibold text-xl tracking-tight"
    >
      SPO 1047
    </span>
  </div>
  <div
    className="w-full block flex-grow lg:flex lg:items-center lg:w-auto"
  >
    <div
      className="text-sm lg:flex-grow"
    >
      <a
        className="block mt-4 lg:inline-block lg:mt-0 text-gray-600 hover:text-gray-400 mr-4"
        href="#responsive-header"
      >
        Beranda
      </a>
      <a
        className="block mt-4 lg:inline-block lg:mt-0 text-gray-600 hover:text-gray-400 mr-4"
        href="#responsive-header"
      >
        Profil
      </a>
    </div>
  </div>
  <div
    className="flex items-center flex-wrap"
  >
    <img
      alt=""
      className="h-8 w-8 rounded-full border-2 border-gray-400"
      src="https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80"
    />
    <p
      className="ml-3 text-sm text-gray-600"
    >
      User
    </p>
  </div>
</nav>
`;
